client/src/redux/sagas/datafiles.sagas.js

```diff
@@ -826,22 +826,88 @@ export function* trashFiles(action) {
   yield call(action.payload.reloadCallback);
 }
 
-export function* trashFile(system, path, id) {
+export function* trashFile(system, path) {
   yield put({
     type: 'DATA_FILES_SET_OPERATION_STATUS_BY_KEY',
-    payload: { status: 'RUNNING', key: id, operation: 'trash' },
+    payload: { status: 'RUNNING', key: system + path, operation: 'trash' },
   });
   try {
     yield call(trashUtil, 'tapis', 'private', system, path);
 
     yield put({
       type: 'DATA_FILES_SET_OPERATION_STATUS_BY_KEY',
-      payload: { status: 'SUCCESS', key: id, operation: 'trash' },
+      payload: { status: 'SUCCESS', key: system + path, operation: 'trash' },
     });
   } catch (e) {
     yield put({
       type: 'DATA_FILES_SET_OPERATION_STATUS_BY_KEY',
-      payload: { status: 'ERROR', key: id, operation: 'trash' },
+      payload: { status: 'ERROR', key: system + path, operation: 'trash' },
+    });
+  }
+}
+
+export async function emptyUtil(api, scheme, system, path) {
+  const url = `/api/datafiles/${api}/delete/${scheme}/${system}${path}/`;
+  const method = 'PUT';
+  return fetchUtil({
+    url,
+    method,
+    body: JSON.stringify({}),
+  });
+}
+
+export function* watchEmpty() {
+  yield takeLeading('DATA_FILES_EMPTY', emptyFiles);
+}
+
+export function* emptyFiles(action) {
+  const emptyCalls = action.payload.src.map((file) => {
+    return call(emptyFile, file.system, file.path);
+  });
+  try {
+    const { result } = yield race({
+      result: all(emptyCalls),
+      cancel: take('DATA_FILES_MODAL_CLOSE'),
+    });
+    yield put({
+      type: 'ADD_TOAST',
+      payload: {
+        message: `${
+          result.length > 1 ? `${result.length} files` : 'File'
+        } deleted `,
+      },
+    });
```

Comment:
@edmondsgarrett Do we still want to handle cases where `result` contains `ERR` responses from the `emptyFile` function? Good to handle the case where the `race` fails for some reason, but I think we can still handle more granular errors within the `try` block like you had before. Apologies if I made it seem like we should remove it altogether

Reply:
Ah, I hadn't considered that! I will make a design thread on Slack about what the text should be for the `catch` case, and I'll add that `else` back in.